includes/discovery/mempools/hikvision-ds.inc.php

```diff
@@ -26,9 +26,9 @@
 if ($device['os'] === 'hikvision-ds') {
     echo 'hikvision-ds:';
 
-    $usage = snmp_get($device, 'memSize.0', '-Ovq', 'HIK-DEVICE-MIB');
-    if ($usage) {
+        $index = 0;
+        $type = 'Generic';
```

Comment:
type should not be "generic", hikvision-ds is probably best.

Reply:
I'll get this taken care of, along with a similar issue for the storage discovery.